### PR TITLE
Add debugging macro and improve other similar macros

### PR DESCRIPTION
**Pull request by keryell**

### CODE REVIEW

File: include/CL/sycl/detail/global_config.hpp [hunk 1]
```diff
@@ -54,22 +54,34 @@
 // But the triSYCL OpenCL implementation is actually based on Boost.Compute
 #include <boost/compute.hpp>
 /// A macro to keep some stuff in OpenCL mode
-#define TRISYCL_SKIP_OPENCL(x) x
+// Use variadic macros so we can have straw commas in the argument
```
Comment (airlied) [inline include/CL/sycl/detail/global_config.hpp]: stray?

File: include/CL/sycl/detail/global_config.hpp [hunk 2]
```diff
@@ -54,22 +54,34 @@
...
-#define TRISYCL_SKIP_OPENCL(x)
+#define TRISYCL_OPENCL_ONLY(...)
 #endif
 
 #ifdef TRISYCL_DEVICE
 /// Define the SYCL marker according to specification
 #define __SYCL_DEVICE_ONLY__
 // To keep something when in device mode
-#define TRISYCL_DEVICE_ONLY(X) X
+// Use variadic macros so we can have straw commas in the argument
```
Comment (airlied) [inline include/CL/sycl/detail/global_config.hpp]: stray?

File: include/CL/sycl/detail/global_config.hpp [hunk 3]
```diff
@@ -54,22 +54,34 @@
...
+#define TRISYCL_DEVICE_ONLY(...) __VA_ARGS__
 #else
 // Do not keep when not in device mode
-#define TRISYCL_DEVICE_ONLY(X)
+#define TRISYCL_DEVICE_ONLY(...)
 #endif
 
+#ifdef TRISYCL_DEBUG
+// To keep something when in debug mode only
+// Use variadic macros so we can have straw commas in the argument
```
Comment (airlied) [inline include/CL/sycl/detail/global_config.hpp]: one more stray?

File: tests/Makefile [hunk 4]
```diff
@@ -34,7 +34,7 @@ CL_CXX = $(LLVM_BUILD_DIR)/bin/clang++
 LDLIBS += -lpthread
 
 # To use OpenCL, set the TRISYCL_OPENCL flag
-CXXFLAGS += -DTRISYCL_OPENCL
+#CXXFLAGS += -DTRISYCL_OPENCL
```
Comment (airlied) [inline tests/Makefile]: is this intentionally part of this? I don't see it addressed.

File: include/CL/sycl/detail/global_config.hpp (hunk 1 above)
Comment (keryell) [reply include/CL/sycl/detail/global_config.hpp]: Ah yes. Thank you for improving my English :-)

File: tests/Makefile (hunk 4 above)
Comment (keryell) [reply tests/Makefile]: No. :-(

**Pull request merged**